edge-hub/core/src/Microsoft.Azure.Devices.Edge.Hub.Core/DeviceScopeIdentitiesCache.cs

```diff
@@ -188,6 +197,67 @@ public async Task<Option<ServiceIdentity>> GetServiceIdentity(string id)
             return await this.GetServiceIdentityInternal(id);
         }
 
+        public async Task<string> VerifyServiceIdentityAuthChainState(string id, bool isNestedEdgeEnabled, bool refreshCachedIdentity)
+        {
+            if (isNestedEdgeEnabled)
+            {
+                return await this.VerifyServiceIdentityAuthChainState(id, refreshCachedIdentity);
+            }
+            else
+            {
+                await this.VerifyServiceIdentityState(id, refreshCachedIdentity);
+                return string.Empty;
+            }
+        }
+
+        async Task VerifyServiceIdentityState(string id, bool refreshCachedIdentity = false)
+        {
+            if (refreshCachedIdentity)
+            {
+                await this.RefreshServiceIdentityInternal(id, this.edgeDeviceId, !refreshCachedIdentity);
+            }
+
+            Option<ServiceIdentity> serviceIdentity = await this.GetServiceIdentity(id);
+
+            this.VerifyServiceIdentity(serviceIdentity.Expect(() =>
+            {
+                Events.VerifyServiceIdentityFailure(id, "Device is out of scope.");
+                return new DeviceInvalidStateException("Device is out of scope.");
+            }));
+        }
+
+        async Task<string> VerifyServiceIdentityAuthChainState(string id, bool refreshCachedIdentity = false)
+        {
+            if (refreshCachedIdentity)
+            {
+                Events.RefreshingServiceIdentity(id);
+
+                var authChainTry = await this.serviceIdentityHierarchy.TryGetAuthChain(id);
+                var actorDeviceId = AuthChainHelpers.GetAuthParent(id, authChainTry.Ok());
```

Comment:
Looks incorrect. `actorDeviceId` should always be `edgeDeviceId`, not parent.

Reply:
I renamed the variable to onBehalfOfDeviceId because that's what it actually is.